splash/browser_tab.py

```diff
@@ -870,6 +870,23 @@ def _post_mouse_event(self, type, button, x, y):
         event = QMouseEvent(type, point, q_button, buttons, modifiers)
         QApplication.postEvent(self.web_page, event)
 
+    def send_keys(self, text, key_type=None):
+        """
+        Send key events to webpage
+        :param text string to be sent as key strokes
+        :param key_type string representing the pressed key
+        :return: None
+        """
+        key_type = key_type or 'unknown'
+        # http://doc.qt.io/qt-5/qt.html#Key-enum
+        key_type = getattr(Qt, 'Key_%s' % key_type, Qt.Key_unknown)
+        self._send_keys(text, key_type)
+
+    def _send_keys(self, text, key_type):
+        modifiers = QApplication.keyboardModifiers()
+        event = QKeyEvent(QEvent.KeyPress, key_type, modifiers, text)
+        QApplication.postEvent(self.web_page, event)
```

Comment:
As an extra, this is the documentation on QKeyEvent

```
int QKeyEvent::key() const

Returns the code of the key that was pressed or released.

See Qt::Key for the list of keyboard codes. These codes are independent of the
underlying window system. Note that this function does not distinguish between
capital and non-capital letters, use the text() function (returning the Unicode
text the key generated) for this purpose.

A value of either 0 or Qt::Key_unknown means that the event is not the result of
a known key; for example, it may be the result of a compose sequence, a keyboard
macro, or due to key event compression.

See also Qt::WA_KeyCompression.
```
http://doc.qt.io/qt-5/qkeyevent.html#key